server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -446,15 +422,11 @@ public DataStream unsafeRollover(Index writeIndex, long generation, boolean time
             indexMode = null;
         }
 
-        List<Index> backingIndices = new ArrayList<>(indices);
+        List<Index> backingIndices = new ArrayList<>(this.backingIndices.indices);
         backingIndices.add(writeIndex);
-        return copy().setIndices(backingIndices)
-            .setGeneration(generation)
-            .setReplicated(false)
```

Comment:
I think it should not be set and it's a bug when we do it. Because when we set it we silently change an attribute of the data stream instead of throwing an error that you cannot do this. 

So +1 on removing it :). Just wondering what is the best way to approach it.

Reply:
I don't think it really is a bug right now, as `replicated` is `false` anyway when we reach this method (except for the forecast use-case, but the `replicated` field seems irrelevant there). If we would start using this "unsafe" method elsewhere, I agree it would be more of a bug. I'm in favour of just removing it in this PR.